GrandCentralBoard/Widgets/Bonus/Source/PeopleWithBonususFetchController.swift

```diff
@@ -0,0 +1,134 @@
+//
+//  BonusFetchController.swift
+//  GrandCentralBoard
+//
+//  Created by Rafal Augustyniak on 07/04/16.
+//  Copyright © 2016 Oktawian Chojnacki. All rights reserved.
+//
+
+import Foundation
+import GrandCentralBoardCore
+
+
+private let kPreferredPeopleCount = 10
+
+enum PeopleWithBonususFetchControllerError : ErrorType {
+    case IncorrectEmailAddress
+    case Cancelled
+    case Unknown
+}
+
+
+final class PeopleWithBonususFetchController {
+
+    private let requestSender: RequestSender
+
+    init(requestSender: RequestSender) {
+        self.requestSender = requestSender
+    }
+
+    func fetchPeopleWithBonuses(completionBlock: (Result<[Person]>) -> Void) {
+        fetchPeopleWithBonuses(startingFromDate: NSDate.init(), fetchedBonuses: [], completionBlock: { [weak self] result in
+            guard let strongSelf = self else { return }
+
+            switch result {
+            case .Success(let people):
+                strongSelf.fetchAvatarsForPeople(people, completionBlock: { result in
+                    switch result {
+                    case .Success(let people):
+                        completionBlock(.Success(people))
+                    case .Failure(let error):
+                        completionBlock(.Failure(error))
+                    }
+                })
+            case .Failure(let error):
+                completionBlock(.Failure(error))
+            }
+
+        })
+    }
+
+    private func fetchPeopleWithBonuses(startingFromDate date: NSDate = NSDate.init(), fetchedBonuses: [Bonus], completionBlock: (Result<[Person]>) -> Void) {
+        let take = 100
+        let requestTemplate = TimestampableRequestTemplate<BonusesRequestTemplate>.init(requestTemplate: BonusesRequestTemplate(), date: date, take: take)
+        
+        requestSender.sendRequestForRequestTemplate(requestTemplate) { [weak self] result in
+            guard let strongSelf = self else {
+                completionBlock(.Failure(PeopleWithBonususFetchControllerError.Cancelled))
+                return
+            }
+
+            switch result {
+            case .Success(let bonuses):
+                var allBonuses: [Bonus] = fetchedBonuses.reverse()
+                allBonuses.appendContentsOf(bonuses)
+
+                let people = allBonuses.reduce(Set<Person>(), combine: { people, bonus in
```

Comment:
it could be maybe moved to extension for `CollectionType where Element == Bonus` (not sure, sth like that)